tasks/eth/base-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,293 @@
+        require(configToCheck.batchInbox() == batchInboxAddress, "1600");
+
+        // Check _addresses
+        require(configToCheck.l1CrossDomainMessenger() == proxies.L1CrossDomainMessenger, "1700");
+        require(configToCheck.l1CrossDomainMessenger().code.length != 0, "1740");
+
+        // L1CrossDomainMessenger is a ResolvedDelegateProxy and that has no getters, so we hardcode some info needed here.
+        AddressManager addressManager = AddressManager(0xdE1FCfB0851916CA5101820A69b13a4E276bd81F); // https://github.com/ethereum-optimism/superchain-registry/blob/4c005f16ee1b100afc08a35a2e418d849bea044a/superchain/extra/addresses/mainnet/op.json#L2
+        address l1xdmImplementation = addressManager.getAddress("OVM_L1CrossDomainMessenger");
+        require(l1xdmImplementation.code.length != 0, "1750");
+
+        require(configToCheck.l1ERC721Bridge() == proxies.L1ERC721Bridge, "1800");
+        require(configToCheck.l1ERC721Bridge().code.length != 0, "1801");
+        require(EIP1967Helper.getImplementation(configToCheck.l1ERC721Bridge()).code.length != 0, "1802");
+
+        require(configToCheck.l1StandardBridge() == proxies.L1StandardBridge, "1900");
+        require(configToCheck.l1StandardBridge().code.length != 0, "1901");
+        require(EIP1967Helper.getImplementation(configToCheck.l1StandardBridge()).code.length != 0, "1902");
+
+        require(configToCheck.optimismPortal() == proxies.OptimismPortal, "2100");
+        require(configToCheck.optimismPortal().code.length != 0, "2101");
+        require(EIP1967Helper.getImplementation(configToCheck.optimismPortal()).code.length != 0, "2102");
+
+        require(configToCheck.optimismMintableERC20Factory() == proxies.OptimismMintableERC20Factory, "2200");
+        require(configToCheck.optimismMintableERC20Factory().code.length != 0, "2201");
+        require(EIP1967Helper.getImplementation(configToCheck.optimismMintableERC20Factory()).code.length != 0, "2202");
+
+        require(configToCheck.disputeGameFactory() == proxies.DisputeGameFactory, "2300");
+        require(configToCheck.disputeGameFactory().code.length != 0, "2301");
+        require(EIP1967Helper.getImplementation(configToCheck.disputeGameFactory()).code.length != 0, "2302");
+
+        // Ensure that the old l2outputoracle slot is cleared
+        bytes32 l2OutputOracleSlot =
+            vm.load(address(configToCheck), bytes32(uint256(keccak256("systemconfig.l2outputoracle")) - 1));
+        require(address(uint160(uint256(l2OutputOracleSlot))) == address(0), "2400");
+    }
+
+    function checkOptimismPortal() internal view {
+        console.log("Running chain assertions on the OptimismPortal");
+
+        // NOTE: proxies.OptimismPortal2 == proxies.OptimismPortal
+        require(proxies.OptimismPortal.code.length != 0, "5700");
+        require(EIP1967Helper.getImplementation(proxies.OptimismPortal).code.length != 0, "5701");
+
```

Comment:
should this be an assertion, e.g. `require(address(proxies.OptimismPortal2) == address(proxies.OptimismPortal), "msg here")`

Reply:
added a sanity check here